File: 0001-0100/026-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.cc
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0;
        for (int j = i+1; j < nums.size(); ++j) {
            if (nums[i] != nums[j]) {
                nums[++i] = nums[j];
            }
        }
        return i+1;
    }
};

int main() {
    Solution s;
    vector<int> nums = {1, 1, 2};
    cout << s.removeDuplicates(nums) << endl;
}

File: leetcode/0001-0100/026-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.js
// 26. Remove Duplicates from Sorted Array
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/description/

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  let i = 0;
  let j = 0;

  while (i < nums.length) {
    if (nums[i] !== nums[i + 1]) {
      nums[j++] = nums[i];
    }
    i++;
  }

  return j;
};

const nums = [1,1,2];
console.info(removeDuplicates(nums));
console.info(nums);
